services/streammanager/googlecloudfunction/googlecloudfunction.go

```diff
@@ -0,0 +1,261 @@
+package cloudfunctions
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"net/http"
+	"reflect"
+	"regexp"
+	"strings"
+
+	"google.golang.org/api/cloudfunctions/v1"
+	"google.golang.org/api/googleapi"
+	"google.golang.org/api/idtoken"
+
+	"google.golang.org/api/option"
+
+	"github.com/tidwall/gjson"
+
+	"github.com/rudderlabs/rudder-go-kit/logger"
+	backendconfig "github.com/rudderlabs/rudder-server/backend-config"
+	"github.com/rudderlabs/rudder-server/services/streammanager/common"
+)
+
+type Config struct {
+	Credentials            string `json:"credentials"`
+	FunctionEnvironment    string `json:"functionEnvironment"`
+	RequireAuthentication  bool   `json:"requireAuthentication"`
+	GoogleCloudFunctionUrl string `json:"googleCloudFunctionUrl"`
+	FunctionName           string `json:"functionName"`
+}
+
+type Client struct {
+	service *cloudfunctions.Service
+	opts    common.Opts
+}
+
+var pkgLogger logger.Logger
+
+func init() {
+	pkgLogger = logger.NewLogger().Child("streammanager").Child("GoogleCloudFunction")
+}
+
+type GoogleCloudFunctionProducer struct {
+	client *Client
+	config *Config
+}
+
+// NewProducer creates a producer based on destination config
+func NewProducer(destination *backendconfig.DestinationT, o common.Opts) (*GoogleCloudFunctionProducer, error) {
+	var config Config
+	jsonConfig, err := json.Marshal(destination.Config)
+	if err != nil {
+		return nil, fmt.Errorf("[GoogleCloudFunction] Error while marshalling destination config :: %w", err)
+	}
+	err = json.Unmarshal(jsonConfig, &config)
+	if err != nil {
+		return nil, fmt.Errorf("[GoogleCloudFunction] error  :: error in GoogleCloudFunction while unmarshalling destination config:: %w", err)
+	}
+
+	opts := []option.ClientOption{
+		option.WithoutAuthentication(),
+	}
+
+	if config.RequireAuthentication {
+		opts = []option.ClientOption{
+			option.WithCredentialsJSON([]byte(config.Credentials)),
+		}
+	}
+
+	service, err := generateService(opts...)
+	// If err is not nil then return
+	if err != nil {
+		pkgLogger.Errorf("[GoogleCloudFunction] error  :: %w", err)
```

Comment:
While logging error I don't think you'd need `[GoogleCloudFunction]`. The logger is already created with `GoogleCloudFunction` as child, so this information should be displayed already. Can you confirm it by checking the logs in your local ?